spring-boot-tools/spring-boot-loader-tools/src/main/resources/org/springframework/boot/loader/tools/launch.script

```diff
@@ -116,9 +116,14 @@ PID_FOLDER="$PID_FOLDER/${identity}"
 pid_file="$PID_FOLDER/{{pidFilename:${identity}.pid}}"
 log_file="$LOG_FOLDER/$LOG_FILENAME"
 
-# Determine the user to run as if we are root
-# shellcheck disable=SC2012
-[[ $(id -u) == "0" ]] && run_user=$(ls -ld "$jarfile" | awk '{print $3}')
+# The application will use the user specified via
+if [[ -z "{{runUser}}" ]]; then
+    # Determine the user to run as if we are root
+    # shellcheck disable=SC2012
+    [[ $(id -u) == "0" ]] && run_user=$(ls -ld "$jarfile" | awk '{print $3}')
```

Comment:
What if root is the owner of the jar file? Might be good to print a warning if someone configures the run as user to root?

Reply:
I thought the same but I didn't want to address this in this PR.  As a running an program as root is a potential security problem. We could issues an warning or abort the script. But this would be breaking change for existing users who updated.

I see to possible solutions:

1. Print a simple warning while starting the program.
2. Refuse to start the program if it would run as root unless the user set `runUser=root` explicitly.

WDYT?

Reply:
I think a warning is the way to go (as you stated it might break something otherwise). I think the Apache Webserver does the same (probably a couple of others as well)

Reply:
Ok, I will update the PR but tomorrow. It is already later for me... Ok?

Reply:
I'm not in the position to  (and I never would) demand anything from you :) plus we're in the same time zone ;)